mixins/segment-analytics.ts

```diff
@@ -1,5 +1,27 @@
-import Vue from 'vue'
+import Vue, { ComponentOptions } from 'vue'
 import { Component } from 'vue-property-decorator'
+import { Route } from 'vue-router'
+
+// Patch the Component decorator to work with abstract classes.
+// @ts-ignore is needed for avoiding the error:
+// "Overload signatures must all be exported or non-exported."
+// See https://github.com/microsoft/TypeScript/issues/21566#issuecomment-362462824
+declare module 'vue-property-decorator' {
+  abstract class _Abstract extends Vue {}
+  // @ts-ignore
+  function Component<V extends Vue>(options: ComponentOptions<V> & ThisType<V>): <VC extends typeof _Abstract>(target: VC) => VC;
+  // @ts-ignore
+  function Component<VC extends typeof _Abstract>(target: VC): VC;
+}
+
+/**
```

Comment:
This comment is a docstring, so more information of use should be provided. As it only has `routeName` we can use the information of the documentation:
"The `routeName` would be a unique name for your route that would allow us to identify the page (category+page) regardless of the URL or the Title of the page changing over the time, that will help us to keep our queries the same over time. Examples: support-guides, notebooks-list, etc.

Please ensure the name is unique on each of your pages and follows kebab-case."

Reply:
You're right. Done in #d16da3c